apps/rush-lib/src/plugins/AzureStorageBuildCachePlugin/index.ts

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as path from 'path';
+import { Import, JsonSchema } from '@rushstack/node-core-library';
+import { IRushPlugin } from '../../pluginFramework/IRushPlugin';
+import { RushSession } from '../../pluginFramework/RushSession';
+import { RushConfiguration } from '../../api/RushConfiguration';
```

Comment:
The eventual goal of plugins is that you could have implemented AzureStorageBuildCachePlugin as a separate project (or even a separate repo).  Looking at this import list, it's hard to tell what is actually exported to the public API of `rush-lib` and what isn't.

It looks like you've followed the basic approach of Heft (defining a RushSession and a pluginFramework, etc.), I'd like to see it taken a step further and have these two plugins pulled into separate _projects_, e.g.:

```console
@rushstack/rush-azure-build-cache-plugin   -->    /rush-plugins/rush-azure-build-cache-plugin
@rushstack/rush-s3-build-cache-plugin      -->    /rush-plugins/rush-s3-build-cache-plugin
```

This way you'll know right away if you're accidentally relying on internal pieces of `rush-lib` that wouldn't normally be available to a plugin.

Reply:
Nice catch and thanks for your hints :) These two plugins are on the way